tests/Unit/Core/Util/ConfigurationVariablesLoaderTest.php

```diff
@@ -0,0 +1,128 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Unit\Core\Util;
+
+use org\bovigo\vfs\vfsStream;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Util\ConfigurationVariablesLoader;
+
+class ConfigurationVariablesLoaderTest extends TestCase
+{
+    public function testLoadEnvVariablesSetValuesWhenParametersAreMissing(): void
+    {
+        $envFile = $this->createBasicEnvFile();
+
+        $configurationVariablesLoader = new ConfigurationVariablesLoader($envFile);
+
+        $this->assertSame(
+            [
+                'parameters' => [
+                    'database_host' => 'localhost',
+                    'database_port' => '3306',
+                    'database_user' => 'db_user',
+                    'database_password' => 'db_password',
+                    'database_name' => 'test_db',
+                    'database_prefix' => 'ps',
+                ],
+            ],
+            $configurationVariablesLoader->loadEnvVariables([]));
+    }
+
+    public function testLoadEnvVariablesReturnsEmptyParametersWhenParametersAreEmpty(): void
+    {
+        $envFile = $this->createBasicEnvFile();
+
+        $configurationVariablesLoader = new ConfigurationVariablesLoader($envFile);
+
+        $this->assertSame(
+            [
+                'parameters' => [
+                    'database_host' => 'localhost',
+                    'database_port' => '3306',
+                    'database_user' => 'db_user',
+                    'database_password' => 'db_password',
+                    'database_name' => 'test_db',
+                    'database_prefix' => 'ps',
+                ],
+            ], $configurationVariablesLoader->loadEnvVariables(['parameters' => []]));
+    }
+
+    public function testLoadEnvVariablesReplacedAllowedParameters(): void
+    {
+        $envFile = $this->createEnvFile('PS_DATABASE_PORT=3306
+PS_DATABASE_NAME=test_db
+PS_DATABASE_PASSWORD=db_password
+PS_DATABASE_PREFIX=ps
```

Comment:
```suggestion
        $envFile = $this->createEnvFile('PS_DATABASE_PORT=330642
PS_DATABASE_NAME=test_database
PS_DATABASE_PASSWORD=db_pwd
PS_DATABASE_PREFIX=test
```

Use different values in order to be sure everything is not always the same between tests.